v2/Tutorial_v2.md

```diff
@@ -0,0 +1,731 @@
+    if err != nil {
+      log.Fatalf("Failed to read document: %v", err)
+    }
+    meta2, err := reader.Read()
+    if err != nil {
+      log.Fatalf("Failed to read document: %v", err)
+    }
+    meta3, err := reader.Read()
+    if err != nil {
+      log.Fatalf("Failed to read document: %v", err)
+    }
+
+    keys := []string{meta1.Key, meta2.Key, meta3.Key}
+
+    fmt.Printf("Created documents with keys '%s' in collection '%s' in database '%s'\n", strings.Join(keys, ","), col.Name(), db.Name())
+  }
+  PrintCollection(db)
+}
+
+func PrintCollection(db arangodb.Database) {
+  var err error
+  var cursor arangodb.Cursor
+
+  querystring := "FOR doc IN users LIMIT 10 RETURN doc"
+
+  cursor, err = db.Query(nil, querystring, nil)
+
+  if err != nil {
+    log.Fatalf("Query failed: %v", err)
+  }
+
+  defer cursor.Close()
+
+  for {
+    var doc User
+    var metadata arangodb.DocumentMeta
+
+    metadata, err = cursor.ReadDocument(nil, &doc)
+
+    if shared.IsNoMoreDocuments(err) {
+      break
+    } else if err != nil {
+      log.Fatalf("Doc returned: %v", err)
+    } else {
+      fmt.Print("Dot doc ", metadata, doc, "\n")
+    }
+  }
+}
+```
+
+## API Design
+
+### Concurrency
+
+All functions of the driver are strictly synchronous. They operate and only
+return a value (or error) when they are done.
+
+If you want to run operations concurrently, use a `go` routine. All objects in
+the driver are designed to be used from multiple concurrent go routines,
+except `Cursor`.
+
+All database objects (except `Cursor`) are considered static. After their
+creation, they don't change. For example, after creating a `Collection` instance,
+you can remove the collection, but the (Go) instance will still be there. Calling
+functions on such a removed collection will, of course, fail.
+
+### Structured error handling & wrapping
+
+All functions of the driver that can fail return an error value. If that value
+is not `nil`, the function call is considered to have failed. In that case, all
+other return values are set to their zero values.
+
+All errors are structured using error-checking functions named
+`Is<SomeErrorCategory>`. For example, `IsNotFound(error)` returns true if the
+given error is of the category "not found". There can be multiple internal error
+codes that all map onto the same category.
+
+All errors returned from any function of the driver (either internal or exposed)
+wrap errors using the `WithStack` function. This can be used to provide detailed
+stack traces in case of an error. All error-checking functions use the `Cause`
+function to get the cause of an error instead of the error wrapper.
+
+### Context-aware
+
+All functions of the driver that involve some kind of long-running operation or
+support additional options are not given as function arguments have a
+`context.Context` argument. This enables you to cancel running requests, pass
+timeouts/deadlines, and pass additional options.
+
+## Connection management
```

Comment:
Needs to be updated for v2 (uses `http` but this isn't even imported in any of the other examples)